From 6ce4e756d6cf5a4cc5539a57b93fcacbfb7654a6 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Sat, 14 May 2016 16:54:32 -0700
Subject: [PATCH] Revert "ls: only display directory headers when more than one
 directory is specified"

This reverts commit b107489bf2cda579ca53551206270b9eee80c059.
---
 ls.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/ls.c b/ls.c
index a59cc0a..55dfe40 100644
--- a/ls.c
+++ b/ls.c
@@ -53,7 +53,6 @@ static int Uflag = 0;
 static int uflag = 0;
 static int first = 1;
 static char sort = 0;
-static size_t ds = 0;
 
 static void ls(const char *, const struct entry *, int);
 
@@ -278,7 +277,7 @@ lsdir(const char *path, const struct entry *dir)
 	if (!Uflag)
 		qsort(ents, n, sizeof(*ents), entcmp);
 
-	if (ds > 1 && (path[0] || dir->name[0] != '.'))
+	if (path[0] || dir->name[0] != '.')
 		printf("%s:\n", dir->name);
 	for (i = 0; i < n; i++)
 		output(&ents[i]);
@@ -369,7 +368,7 @@ int
 main(int argc, char *argv[])
 {
 	struct entry ent, *dents, *fents;
-	size_t i, fs;
+	size_t i, ds, fs;
 
 	ARGBEGIN {
 	case '1':
-- 
2.8.1

